docs/ide/mcp.md

```diff
@@ -0,0 +1,166 @@
+ms.date: 5/8/2025
+ms.topic: overview
+author: aayim
+ms.author: meghaanand
+ms.manager: mijacobs
+ms.subservice: ai-tools
+ms.collection: ce-skilling-ai-copilot
+monikerRange: '>= vs-2022'
+---
+# What is MCP?
+
+Model Context Protocol (MCP) is an open standard that enables AI models to interact with external tools and services through a unified interface. In Visual Studio, MCP support enhances GitHub Copilot's agent mode by allowing you to connect any MCP-compatible server to your agentic coding workflow. This article guides you through setting up MCP servers and using tools with agent mode in Visual Studio.
+
+## Prerequisites
++ [Visual Studio 2022 version 17.14](/visualstudio/releases/2022/release-history) and later offer Copilot Agent.
+    + In the initial 17.14 release, you will need to first enable `Agent Mode` in `Tools > Options > GitHub > Copilot`.
++ [Sign in to Visual Studio using a GitHub account](work-with-github-accounts.md) with [Copilot access](https://docs.github.com/en/copilot/about-github-copilot/what-is-github-copilot#getting-access-to-copilot). <br/>
+    + You can use [GitHub Copilot for Free](copilot-free-plan.md). Sign up and leverage AI to code faster and more efficiently.
++ Enable [agent mode in GitHub Copilot Chat](copilot-agent.md).
+
+## How do MCP and Visual Studio extend GitHub Copilot's agent?
+
+* MCP clients (like Visual Studio) connect to MCP servers and request actions on behalf of the AI model
+* MCP servers provide one or more tools that expose specific functionalities through a well-defined interface.
+* The Model Context Protocol (MCP) defines the message format for communication between clients and servers, including tool discovery, invocation, and response handling
+
+For example, a file system MCP server might provide tools for reading, writing, or searching files and directories. GitHub's MCP server offers tools to list repositories, create pull requests, or manage issues. MCP servers can run locally on your machine or be hosted remotely, and Visual Studio supports both configurations.
+
+By standardizing this interaction, MCP eliminates the need for custom integrations between each AI model and each tool. This allows you to extend your AI assistant's capabilities by simply adding new MCP servers to your workspace. Learn more about the Model Context Protocol specification.
+
+# Quick Start
+## Configuration example with GitHub MCP server
+
+1. Create a new file: `<SOLUTIONDIR>\.vs\mcp.json`
+2. Paste the following contents into the `mcp.json` file
+
+```markdown
+{
+  "inputs": [
+    {
+      "type": "promptString",
+      "id": "github_token",
+      "description": "GitHub Personal Access Token",
+      "password": true
+    }
+  ],
+  "servers": {
+    "github": {
+      "command": "npx",
+      "args": [ "-y", "@modelcontextprotocol/server-github" ],
+      "env": {
+        "GITHUB_PERSONAL_ACCESS_TOKEN": "GITHUB_PAT"
```

Comment:
Are we sure this is correct? This is different from the official docs, and given that we're asking for an input above, I'd expect this to reference the input - that, or there doesn't need to be an input, if users are going to directly specify the PAT in the config, but this appears to be a hybrid of the two

Reply:
great catch, just fixed it